src/tcache.c

```diff
@@ -357,25 +357,18 @@ tcache_destroy(tsd_t *tsd, tcache_t *tcache)
 		tcache_bin_t *tbin = &tcache->tbins[i];
 		tcache_bin_flush_small(tsd, tcache, tbin, i, 0);
 
-		if (config_stats && tbin->tstats.nrequests != 0) {
-			arena_bin_t *bin = &arena->bins[i];
-			malloc_mutex_lock(tsd_tsdn(tsd), &bin->lock);
-			bin->stats.nrequests += tbin->tstats.nrequests;
-			malloc_mutex_unlock(tsd_tsdn(tsd), &bin->lock);
-		}
+		if (config_stats) {
+			assert(tbin->tstats.nrequests == 0);
+                }
```

Comment:
This line's indentation is spaces rather than tabs, which is why the diff doesn't align as expected.